lib/motions/general-motions.coffee

```diff
@@ -149,10 +149,16 @@ class MoveLeft extends Motion
 class MoveRight extends Motion
   operatesInclusively: false
 
-  moveCursor: (cursor, count=1) ->
+  # MoveRight can be combined with Delete, which will indicate it by the presence of options.deleting
+  moveCursor: (cursor, count=1, options={}) ->
     _.times count, =>
-      cursor.moveRight() unless cursor.isAtEndOfLine()
-      cursor.moveRight() if settings.wrapLeftRightMotion() and cursor.isAtEndOfLine()
+      goingToNextLine = atom.config.get('vim-mode.wrapLeftRightMotion')
+
+      if not cursor.isAtEndOfLine() # the line is not empty
+        cursor.moveRight()
+        goingToNextLine = false if options.deleting # when deleting, wrapLeftRightMotion only works on an empty line
```

Comment:
My general sense is that we should make Atom feel like vim, but not necessarily reimplement all of vim's inconsistencies. The `deleting` option that you added makes complete sense as a fix for this particular bug, but it bothers me that a specific motion like `MoveRight` would have knowledge of a particular type of operator (`deleting`) and take on a different behavior when used with those operators.

Regarding the findings that you shared above, are there any operators whose behavior would *badly* diverge from their behavior in vim if we were to handle `wrapLeftRightMotion` uniformly for all operators (by replacing `if options.deleting` with `if @vimState.mode is 'operator-pending'`)?